src/Configuration/src/CloudFoundry/CloudFoundryConfigurationProvider.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace Steeltoe.Extensions.Configuration.CloudFoundry;
 
-public class CloudFoundryConfigurationProvider : ConfigurationProvider
+public sealed class CloudFoundryConfigurationProvider : ConfigurationProvider
```

Comment:
I know we want to be as restrictive as possible, but I don't see this necessary - the idea seems to be that users of the library can over ride the Load method for example - which cannot be anymore.

Reply:
Counterpoint: Cloud Foundry configuration (vcap application/services) should be quite stable by now, and we should be basically be handling it appropriately... With the way layering works in IConfiguration, if they want to override any value coming from the cf environment, wouldn't they put another config provider after this one? I don't know that I have a strong opinion on whether or not this should be sealed, but between that form of override and the ability to specify your own `ICloudFoundrySettingsReader`, what would an end user want to override here?

Reply:
I've made this sealed because the class doesn't seem to be designed for inheritance. While the `Load` method was indeed overridable, all other methods were private, which it delegates to. So for a consumer to override, they must re-implement all the private logic.

If we envision consumers to inherit from this class, we should change the private methods into protected virtual methods that can be used as reusable building blocks. Those methods then need to be properly documented, validate their input parameters, guard internal state etc. Along with that, we need to expose private fields, assert they are valid etc.

Reply:
The Idea is that they override the load method, while keeping the benefit of the constructors we provide. Otherwise they will have to implement the interface from scratch - but to Tims point may be we have enough ability to customize via ICloudfoundrySettingsReader.

Reply:
Have a look at Microsoft's ConfigurationProviders ... e.g. Json...., etc.